ql/src/java/org/apache/hadoop/hive/ql/parse/CalcitePlanner.java

```diff
@@ -683,20 +681,17 @@ Operator genOPTree(ASTNode ast, PlannerContext plannerCtx) throws SemanticExcept
             // Wrap all other errors (Should only hit in tests)
             throw new SemanticException(e);
           } else {
-            reAnalyzeAST = true;
+            String strategies = conf.getVar(ConfVars.HIVE_QUERY_REEXECUTION_STRATEGIES);
```

Comment:
I think it would be more readable what/why are you doing this if there would be a method like move this into some method like `ReCompilationPlugin#isEnabled(conf)`  or something like that

Reply:
Moved the check to the `ReExecDriver.checkHookConfig`